src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -34005,295 +37604,358 @@ class Program
-                        Operand:
-                          ICollectionExpressionOperation (0 elements, ConstructMethod: null) (OperationKind.CollectionExpression, Type: MyCollection[]) (Syntax: '[]')
-                            Elements(0)
-                """);
+        // [Obsolete] attribute is ignored when checking for Add for conversion.
+        [Fact]
+        public void AddMethod_Obsolete_02()
+        {
+            string source = """
+                using N;
+                using System;
+                using System.Collections;
+                using System.Collections.Generic;
+                class MyCollection<T> : IEnumerable
+                {
+                    private readonly List<T> _list = new();
+                    IEnumerator IEnumerable.GetEnumerator() => _list.GetEnumerator();
+                    internal void __AddInternal(T t) { _list.Add(t); }
+                }
+                static class Extensions
+                {
+                    [Obsolete("do not use", error: true)]
+                    public static void Add<T>(this MyCollection<T> collection, string s) => throw null;
+                }
+                namespace N
+                {
+                    static class Extensions
+                    {
+                        public static void Add<T>(this MyCollection<T> collection, T t) { collection.__AddInternal(t); }
+                    }
+                }
+                class Program
+                {
+                    static void Main()
+                    {
+                        int x = 1;
+                        int[] y = [2, 3];
+                        MyCollection<object> z = [x, ..y];
+                        z.Report();
+                    }
+                }
+                """;
+            CompileAndVerify([source, s_collectionExtensions], expectedOutput: "[1, 2, 3], ");
         }
 
-        [WorkItem("https://github.com/dotnet/roslyn/issues/72461")]
         [Fact]
-        public void Add_ParamsArray_07()
+        public void AddMethod_Unsafe()
         {
             string sourceA = """
                 using System.Collections;
                 using System.Collections.Generic;
-                struct MyCollection : IEnumerable
+                class MyCollection<T> : IEnumerable
                 {
-                    private List<MyCollection?> _list;
-                    public void Add(params MyCollection?[] x) => GetList().AddRange(x);
-                    public IEnumerator<MyCollection?> GetEnumerator() => GetList().GetEnumerator();
+                    private readonly List<T> _list = new();
+                    public IEnumerator<T> GetEnumerator() => _list.GetEnumerator();
                     IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
-                    private List<MyCollection?> GetList() => _list ??= new();
+                    unsafe public void Add(void* p) => throw null;
```

Comment:
Let's have a variant where this is the closest extension method, but there's a better (safe) extension method further away.